### PR TITLE
Benchmark

### PR DESCRIPTION
**Pull request by GirZ0n**
**Changes**
- Added an option to ignore empty lines when calculating the number of lines of code.
- Changed the behavior of `submissions_metrics_statistics.py` a bit. Now you have to specify the appropriate flag to calculate some metric.
- Moved `time_compare.py` into `evaluation` module and adapted it to the new changes with Hyperstyle and Qodana/ evaluation.

### CODE REVIEW

File: analysis/src/python/data_analysis/statistics/submissions_metrics_statistics.py [hunk 1]
```diff
@@ -1,52 +1,141 @@
...
-from analysis.src.python.utils.logging_utils import configure_logger
-from analysis.src.python.data_analysis.utils.stats_utils import calculate_code_lines_count, \
-    calculate_code_symbols_count, calculate_issues_count
 
 
-def get_submission_statistics(submissions_with_issues_path: str, submissions_statistics_path: str):
-    """ Calculate submissions metrics such number of code lines, symbols, issues. """
+def get_submission_statistics(
+    submissions: pd.DataFrame,
+    code_lines_count: Optional[Literal['all', 'ignore_empty_lines']] = None,
```
Comment (nbirillo) [inline analysis/src/python/data_analysis/statistics/submissions_metrics_statistics.py:19]: please, add an enum or something like this for 'all', 'ignore_empty_lines'

File: analysis/src/python/evaluation/benchmark/benchmark.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+            return None
+
+    @classmethod
+    def available_values(cls) -> Set[str]:
+        return {cls.HYPERSTYLE.value, cls.QODANA.value}
+
+
+def measure_run_time(f: Callable[[], Any], repeat: int, output_path: Path, parser: Callable[[Path], Any]) -> float:
+    repeat_times = []
+    for i in range(1, repeat + 1):
```
Comment (nbirillo) [inline analysis/src/python/evaluation/benchmark/benchmark.py:44]: range(repeat)? and logger.info(f'Time measuring attempt={i + 1}')?

File: analysis/src/python/evaluation/benchmark/benchmark.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+    language_version = get_language_version(row[SubmissionColumns.LANG.value])
+    input_path = create_directory(config.tmp_path / 'input', clear=True)
+    output_path = create_directory(config.tmp_path / 'output', clear=True)
+
+    save_solutions_to_files(row.to_frame().T, language_version, input_path, config.with_template)
+
+    command = config.build_command(input_path, output_path, language_version)
+
+    mean_time = measure_run_time(
```
Comment (nbirillo) [inline analysis/src/python/evaluation/benchmark/benchmark.py:95]: probably it is better to add a cli argument to choose what time to check the user wants: mean or median. What do you think?

**Review by nbirillo — CHANGES_REQUESTED** [review]
Great! I have several small comments

File: analysis/src/python/evaluation/benchmark/benchmark.py [hunk 4]
```diff
@@ -35,33 +35,61 @@ def from_value(cls, value: str) -> Optional['Analyzer']:
             return None
 
     @classmethod
-    def available_values(cls) -> Set[str]:
+    def values(cls) -> Set[str]:
         return {cls.HYPERSTYLE.value, cls.QODANA.value}
 
 
-def measure_run_time(f: Callable[[], Any], repeat: int, output_path: Path, parser: Callable[[Path], Any]) -> float:
+class AggregateOption(Enum):
```
Comment (nbirillo) [inline analysis/src/python/evaluation/benchmark/benchmark.py:42]: Can we unify this class with [this](https://github.com/hyperskill/hyperstyle-analyze/blob/main/analysis/src/python/data_analysis/template_errors/models/postprocessing_models.py#L10) class?

**Review by nbirillo — CHANGES_REQUESTED** [review]
One more comment

**Comment by nbirillo** [comment]
@GirZ0n can I check it?

**Comment by GirZ0n** [comment]
@nbirillo, yes, you can. I forgot to re-request a review :sweat_smile:

**Review by nbirillo — APPROVED** [review]
Perfect! Thank you!

**Pull request merged**